integration_tests/src/main/python/join_test.py

```diff
@@ -210,6 +213,48 @@ def do_join(spark):
         return left.join(right, left.key == right.r_key, join_type)
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=_hash_join_conf)
 
+# Test GpuBroadcastHashJoin with right broadcast and no condition
+@validate_execs_in_gpu_plan('GpuBroadcastHashJoinExec')
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', basic_nested_gens + decimal_128_gens, ids=idfn)
+@pytest.mark.parametrize('join_type', ['Left', 'LeftSemi', 'LeftAnti'], ids=idfn)
+def test_right_broadcast_hash_join_ridealong(data_gen, join_type):
```

Comment:
Isn't this a duplicate of the existing `test_broadcast_join_right_table_ridealong` test?  I'm not sure why this PR would be adding tests for joins without any condition.